src/tests/egui_demo.edn

```diff
@@ -0,0 +1,68 @@
+; SPDX-License-Identifier: BSD-3-Clause
+; Copyright © 2022 Fragcolor Pte. Ltd.
+
+(def lorem_ipsum
+  "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.")
+
+(defshards window-with-panels []
+  (UI.Window
+   :Title "Window with Panels"
+   :Contents
+   (-> (UI.TopPanel
+        (->
+         "Top Window Panel" (UI.Label)))
+       (UI.BottomPanel
+        (->
+         "Bottom Window Panel" (UI.Label)))
+       (UI.LeftPanel
+        (->
+         "Left Window Panel" (UI.Label)))
+       (UI.RightPanel
+        (->
+         "Right Window Panel" (UI.Label)))
+       (UI.CentralPanel
+        (->
+         "Central Window Panel" (UI.Label))))))
+
+(defloop ui-wire
+  (GFX.MainWindow
+   :Title "egui Shards Example" :Width 1280 :Height 768 :Debug false
+   :Contents
+   (->
+    (Setup
+     (GFX.DrawQueue) >= .ui-draw-queue
+     (GFX.UIPass .ui-draw-queue) >> .render-steps)
+
+    (| .ui-draw-queue (GFX.ClearQueue))
```

Comment:
It's because (UI) is supposed to pass it's input to its inner shards, in case you want the UI to get the wire input. Otherwise it would be the queue being passed around.

Granted in this sample we don't use the input, but consider:

```clj
(defloop ui-wire
  42 ;; input here
  (GFX.MainWindow
   :Title "egui Shards Example" :Width 1280 :Height 768 :Debug false
   :Contents
   (->
    (Setup
     (GFX.DrawQueue) >= .ui-draw-queue
     (GFX.UIPass .ui-draw-queue) >> .render-steps)

    (| .ui-draw-queue (GFX.ClearQueue))

    ;; here I want to pass 42
    (UI
     .ui-draw-queue
     (-> ...)))))
```

Reply:
Wouldn't you just add it when you need it. No need to do so if it's not used in this test/sample